lib/logstash/plugin_mixins/http_client.rb

```diff
@@ -1,113 +1,107 @@
 # encoding: utf-8
 require "logstash/config/mixin"
+require 'logstash/plugin_mixins/http_client_ssl_configs'
 
 # This module makes it easy to add a very fully configured HTTP client to logstash
 # based on [Manticore](https://github.com/cheald/manticore).
 # For an example of its usage see https://github.com/logstash-plugins/logstash-input-http_poller
 module LogStash::PluginMixins::HttpClient
-  class InvalidHTTPConfigError < StandardError; end
+  ##
+  # Configure the module using the provided config hash. The available options are:
+  # :ssl_deprecated_options [bool]: enable or disable all deprecated SSL configurations
+  # :ssl_standardized_options [bool]: enable or disable all standardized SSL configurations
+  # @return [Self]
+  def self.[](config)
+    validate_module_config!(config)
+    @http_client_module_config = config
```

Comment:
In my reference implementation, I used an [`Adapter`](https://github.com/logstash-plugins/logstash-mixin-http_client/pull/43/files#diff-ea532f763cda4853b848c9fe7053fd21f3008b3155f8b32824add72d3f441465R21), which is a class that inherits from `Module` so it can be instantiated with arguments into a `Module` that in turn can be `include`-d into the plugin class.